_posts/2022-10-09-decentralization-of-zk-rollups.md

```diff
@@ -0,0 +1,192 @@
+4. Economically aligned incentives. Participants should be rewarded equally for upkeep of the network, in order to encourage new participants to enter. 
+5. Liveness. Honest messages should always be included and available to block producers. Chances of halting or lapses in network availability  should be mitigated. 
+
+
+## Upgrade Keys
+
+Upgrade keys are public keys that have been whitelisted by a rollup’s L1 smart contract, and they have two main benefits:
+
+
+
+1. Can fix buggy code, an eventuality that cannot be escaped, as reducing the chance of this happening is very hard. 
+2. Can improve the protocol to update with new technology advancements. This is available on L1s through hard forks.
+
+But, this amount of centralized control over an L2 is not very safe as this entity could change the complete state and protocol. 
+
+How do we mitigate this risk? We have identified some ways to move forward:
+
+Abolish upgrade keys permanently. If there are any changes to be made with the original smart contract, bundle them together and deploy a new smart contract with the changes. This is extremely abrupt and harsh. You will then have to entice the miners and users to organically switch over from the “old” network. You can ease into it by airdropping tokens of the “new” rollup to users of the “old” rollup equivalent to their last checked balance, or by accepting the wrapped L1 tokens from the previous L2 rollup.
+
+Another possible solution is to bake in social consensus to the L1 contract, so that if a smart contract change is to be made, it first needs to be agreed by all the participants of the L2 network. However, this opens up to new problems regarding this type of governance, such as the handling of urgent situations, voter fatigue, and power centralization[^1]<sup>,</sup>[^2]. It would also be advisable to add delays, so if the users don’t like an upgrade, they have time to exit, and move their assets back to the L1. 
+
+
+## Why Proof of Stake (PoS) Might Not Work
+
+Currently, zk-rollups have a centralization in the block creator. To decentralize the block creator, a question is raised on who we decide as the leader / coordinator for a given block. 
+
+There are multiple answers to this question (some we will discuss further in this article): MEV Auction, Burn Auction, Proof of Efficiency and finally Proof of Stake. Most know Proof of Stake from their implementation in L1 blockchains such as Ethereum. However, they can also be used in zk-rollups for leader election, with some teams actively working on this: Polygon Zero, and Starkware. 
+
+The problem with integrating PoS into a zk rollup is that you are unable to fo a hard fork, which provided most of the security guarantees in L1 implementations. This makes it susceptible to DOS and Fake DOS attacks. 
```

Comment:
Why? Can't the staking and voting power be extended in some way, such as getting involved with upgradeability, and go beyond merely proposing blocks?

Reply:
Not sure what you are referring to here? This section does not concern upgradeability.

Reply:
Why is the old chain not accessible? Shouldn't the people who put substantial stake there actively make copies and backup of historical transactions, state, and current updates?